esmvalcore/_recipe.py

```diff
@@ -890,6 +910,34 @@ def _get_preprocessor_task(variables, profiles, config_user, task_name):
             )
             derive_tasks.append(task)
 
+    # special case: fx variable pre-processing
+    for step in ('area_statistics', 'volume_statistics'):
```

Comment:
I think `zonal_statistics` should be added to this list.
```suggestion
    for step in ('area_statistics', 'volume_statistics', 'zonal_statistics'):
```